doc/reference/reference_lua/box_null.rst

```diff
@@ -0,0 +1,138 @@
+.. _box-null:
+
+-------------------------------------------------------------------------------
+                             Constant `box.NULL`
+-------------------------------------------------------------------------------
+
+There are some major problems with using a Lua **nil** value in tuples.
+For example: you can't correctly assess length of sparse array (an array with
+nullable values set to NULL) in case of using Lua **nil** value.
+
+**Example:**
+
+.. code-block:: tarantoolsession
+
+    tarantool> t = {0, nil, 1, 2, nil}
+    ---
+    ...
+
+    tarantool> t
+    ---
+    - - 0
+      - null
+      - 1
+      - 2
+    ...
+
+    tarantool> #t
+    ---
+    - 4
+    ...
+
+Provided example clearly shows a problem with sparse arrays.
+
+.. NOTE::
+
+    Trying to find length for sparse arrays in LuaJIT leads to
+    `undefined behaviour <https://www.lua.org/manual/5.2/manual.html#3.4.6>`_.
+
+To overcome this and some other issues use ``box.NULL`` constant instead of **nil**.
```

Comment:
Since it is unspecified in documentation - you tell me. Or this is the only issue we could have with it?

Reply:
To overcome this issue use ``box.NULL`` constant instead of **nil**. box.NULL here is a placeholder for a **nil** value in tables to preserve key without a value.